android/UtilsPlugin.java

```diff
@@ -33,380 +38,412 @@
-			JSONObject shortcuts = new JSONObject(params);
-			Iterator<?> keys = shortcuts.keys();
-			List<ShortcutInfo> shortcutList = new ArrayList<ShortcutInfo>();
-
-			while (keys.hasNext()) {
-				String key = (String)keys.next();
-				JSONObject data = shortcuts.getJSONObject(key);
-				String title = data.getString("title");
-				logger.log("{utils-native} adding shortcut for : " + key);
-
-				Intent shortcutIntent = _context.getPackageManager().getLaunchIntentForPackage(_context.getPackageName());
-				shortcutIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
-				shortcutIntent.setAction(TeaLeaf.ACTION_SHORTCUT);
-				shortcutIntent.putExtra(TeaLeaf.SHORTCUT_KEY, key);
-
-				ShortcutInfo shortcut = new ShortcutInfo.Builder(_activity, key)
-                                	.setShortLabel(title)
-                                	.setLongLabel(title)
-					.setIcon(Icon.createWithResource(_context, _context.getResources().getIdentifier("shortcut_" +
-						 data.getString("icon"), "drawable", _context.getPackageName())))
-                                	.setIntent(shortcutIntent)
-					.build();
-				shortcutList.add(shortcut);
-			}
-
-			shortcutManager.setDynamicShortcuts(shortcutList);
-		} catch (Exception ex) {
-			logger.log("{utils-native} Exception creating shortcut... " + ex);
-		}
-	}
-
-	public void getInstalledApps(String params) {
-		JSONArray appsList = new JSONArray();
-
-		try {
-			Set<String> appData = new HashSet<String>(Arrays.asList(getSettings().getString(TeaLeafReceiver.INSTALLED_APPS_KEY, "").trim().split(TeaLeafReceiver.PACKAGE_DELIMITER)));
-
-			for (String currApp: appData) {
-				JSONObject currObj = new JSONObject();
-				String[] data = currApp.split(TeaLeafReceiver.INSTALL_TIME_DELIMITER);
-				currObj.put("packageName", data[0]);
-				currObj.put("installTime", data[1]);
-				appsList.put(currObj);
-			}
-		} catch (JSONException e) {
-			logger.log("{utils-native} JSONException on getInstalledApps" + e);
-		} catch (NullPointerException e) {
-			logger.log("{utils-native} NullPointerException on getInstalledApps" + e);
-		} catch (Exception e) {
-			logger.log("{utils-native} Exception on getInstalledApps" + e);
-		}
-
-		EventQueue.pushEvent(new AppsInstalled(appsList));
-	}
-
-	public void isAppInstalled(String params) {
-		boolean available = false;
-
-		PackageManager pm = _context.getPackageManager();
-		try {
-			JSONObject data = new JSONObject(params);
-			String packageName = data.optString("identifier", "");
-
-			pm.getPackageInfo(packageName, PackageManager.GET_ACTIVITIES);
-			available = true;
-		} catch (PackageManager.NameNotFoundException e) {
-			logger.log("{utils-native} app not available for package");
-		} catch (JSONException e) {
-			logger.log("{utils-native} isAppInstalled json exception" + e);
-		}
-
-		EventQueue.pushEvent(new AppFound(available));
-	}
-}
+    Context _context;
```

Comment:
We prefer 2 spaces when using spaces.
Can you please change it to 2 spaces. Thank you